py-scripts/lf_interop_video_streaming.py

```diff
@@ -1050,6 +1050,120 @@ def process_list(self, lst):
+        mode = data["mode"]
+        total_buffer = data["total_buffer"]
+        wait_time_data = data["wait_time_data"]
+        min_video_rate = data["min_video_rate"]
+        avg_video_rate = data["avg_video_rate"]
+        max_video_rate = data["max_video_rate"]
+        total_urls = data["total_urls"]
+        total_err = data["total_err"]
+        rssi_data = data["rssi_data"]
+        tx_rate = data["tx_rate"]
+        max_bytes_rd_list = data["max_bytes_rd_list"]
+        avg_rx_rate_list = data["avg_rx_rate_list"]
+
+        test_input_list = []
+        pass_fail_list = []
+
+        if self.expected_passfail_val or self.csv_name:
+            if not self.expected_passfail_val:
+                res_list = []
+                interop_tab_data = self.json_get('/adb/')["devices"]
+                for client in range(len(device_type[:ci])):
+                    if device_type[client] != 'Android':
+                        res_list.append(username[:ci][client])
+                    else:
+                        for dev in interop_tab_data:
+                            for item in dev.values():
+                                if item['user-name'] == username[:ci][client]:
+                                    res_list.append(item['name'].split('.')[2])
+
+                if self.csv_name is None:
+                    self.csv_name = "device.csv"
+                try:
+                    with open(self.csv_name, mode='r') as file:
+                        reader = csv.DictReader(file)
+                        rows = list(reader)
+                except Exception as e:
+                    logger.error(f"An error occurred while reading the CSV file: {e}")
+                    return
+
+                for device in res_list:
+                    found = False
+                    for row in rows:
+                        if row['DeviceList'] == device and row['Videostreaming URLcount'].strip() != '':
+                            test_input_list.append(row['Videostreaming URLcount'])
+                            found = True
+                            break
+                    if not found:
+                        logging.info(f"Pass Fail Value for Device {device} not found in CSV. Using default value 5")
+                        test_input_list.append(5)
+
+                for i in range(len(test_input_list)):
+                    if float(test_input_list[i]) <= total_urls[:ci][i]:
+                        pass_fail_list.append('PASS')
+                    else:
+                        pass_fail_list.append('FAIL')
+            else:
+                test_input_list = [self.expected_passfail_val for _ in range(len(username[:ci]))]
+                for i in range(len(test_input_list)):
+                    if int(self.expected_passfail_val) <= total_urls[:ci][i]:
+                        pass_fail_list.append("PASS")
+                    else:
+                        pass_fail_list.append("FAIL")
+
+            return {
```

Comment:
Looks like a lot of duplicated code here in returns. Fine for now but we should fix this in the future.